spec/concurrent/executor/cached_thread_pool_spec.rb

```diff
@@ -201,6 +201,30 @@ module Concurrent
         end
       end
 
+      context 'auto terminate' do
+
+        # https://github.com/ruby-concurrency/concurrent-ruby/issues/817
+        # https://github.com/ruby-concurrency/concurrent-ruby/issues/839
+        it 'does not stop shutdown ' do
+          Timeout::timeout(10) do
+            `
```

Comment:
@pitr-ch I re-worked the test and have it running via spec.

I validated that this test on _master_ causes the test to hang.
Unfortunately, I can't seem to get it to gracefully fail even with Timeout. The timeout error raising _Kernel#raise_ does not seem to interrupt it waiting for the subprocess. Not sure how much it matters since we expect the test to pass...

note: You can't use Process#fork on JRuby